lib/common/marker.js

```diff
@@ -0,0 +1,237 @@
+
+    var errs = [];
+
+    if (!strictTotalOrderFieldInMarker(marker)) {
+        errs.push('A marker needs to have a ' + STRICT_TOTAL_ORDER_FIELD +
+            ' property from which a strict total order can be established');
+    }
+
+    if (sort !== undefined) {
+        // Make sure that all fields used to sort the results set
+        // are present in the marker so that a strict total order can be
+        // established by the marker over the sorted results set.
+        if (!sortFieldInMarker(sort, marker)) {
+            errs.push('All sort fields must be present in marker.' +
+                ' Sort fields: ' + sort + '.');
+        }
+    }
+
+    // Conversely, make sure that all marker fields are present in the
+    // sort parameter so that a strict total order can be established by
+    // the marker over the sorted results set.
+    if (!allMarkerFieldsInSort(marker, sort)) {
+        errs.push('All marker keys except ' + STRICT_TOTAL_ORDER_FIELD +
+            ' must be present in the sort parameter.' + ' Sort fields: ' +
+            sort + '.');
+    }
+
+    return errs;
+}
+
+/*
+ * Returns true if the string "sortField" is also one of the marker's object
+ * field.
+ *
+ * For instance, it returns true if:
+ * - sortField is "create_timestamp" and marker is
+ * "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ *
+ * It returns false if:
+ * - sortField is "create_timestamp" and marker is
+ * "{"uuid": "some-uuid"}
+ */
+function sortFieldInMarker(sortField, marker) {
+    assert.string(sortField, 'sortField');
+    assert.object(marker, 'marker');
+
+    // If sortField specifies an order with 'field.ASC' or 'field.DESC',
+    // remove the order and just keep the sort field name.
+    sortField = sortField.split('.')[0];
+
+    return Object.keys(marker).some(function (markerField) {
+        return sortField === markerField;
+    });
+}
+
+/*
+ * Returns true if all fields in marker "marker" are also present in the sort
+ * fields represented by "sort", false otherwise
+ *
+ * For instance, it returns true if:
+ * - marker is "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ * - sort is "create_timestamp"
+ *
+ * It returns false if:
+ * - marker is "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ * - sort is "brand"
+ */
+function allMarkerFieldsInSort(marker, sort) {
+    assert.object(marker, 'marker must be an object');
+    assert.optionalString(sort, 'sort must be a string or undefined');
+
+    sort = sort || '';
+
+    var markerFields = Object.keys(marker);
+    if (markerFields.length === 0) {
+        // A marker object with no property have all its fields
+        // included in "sort", since it has none.
+        return true;
+    }
+
+    // If sortField specifies an order with 'field.ASC' or 'field.DESC',
+    // remove the order and just keep the sort field name.
+    var sortField = sort.split('.')[0];
```

Comment:
Addressed by [the latest commit in this PR's branch](https://github.com/misterdjules/sdc-vmapi/commit/5d65624f5bf76038e74fd8c831bcf56a6f42102f#diff-ecd0ef3f6dd2e7e584474f39b6c31136R188).